crates/wallets/src/utils.rs

```diff
@@ -0,0 +1,145 @@
+use ethers_signers::{HDPath as LedgerHDPath, LocalWallet, TrezorHDPath, WalletError};
+use foundry_config::Config;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
+use eyre::{Context, Result};
+
+use crate::{error::PrivateKeyError, PendingSigner, WalletSigner};
+
+pub fn create_private_key_signer(private_key: &str) -> Result<WalletSigner> {
```

Comment:
can this be a function on the WalletSigner type instead?

Reply:
yep, but it feels like this functions's logic doesn't really belong there tbh as it's more about user input validation then about WalletSigner construction

wdyt?

Reply:
I see, actually keeping it in utils is fine then
a few docs would be great